packages/cacti-plugin-consortium-static/src/main/typescript/plugin-consortium-static.ts

```diff
@@ -0,0 +1,503 @@
+    return webServices;
+  }
+
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    const { log } = this;
+    const pkgName = this.getPackageName();
+
+    if (this.endpoints) {
+      return this.endpoints;
+    }
+    log.info(`Creating web services for plugin ${pkgName}...`);
+    // presence of webAppOptions implies that caller wants the plugin to configure it's own express instance on a custom
+    // host/port to listen on
+
+    const { keyPairPem } = this.options;
+    const consortiumRepo = this.repo;
+
+    const endpoints: IWebServiceEndpoint[] = [];
+    {
+      const options = { keyPairPem, consortiumRepo, plugin: this };
+      const endpoint = new AddNewNodeEndpoint(options);
+      endpoints.push(endpoint);
+      const path = endpoint.getPath();
+      this.log.info(`Instantiated AddNodeEndpoint at ${path}`);
+    }
+    {
+      const options = { keyPairPem, consortiumRepo, plugin: this };
+      const endpoint = new ProcessBroadcastEndpoint(options);
+      endpoints.push(endpoint);
+      const path = endpoint.getPath();
+      this.log.info(`Instantiated AddNodeEndpoint at ${path}`);
+    }
+    {
+      const options = { keyPairPem, consortiumRepo, plugin: this };
+      const endpoint = new GetConsortiumEndpointV1(options);
+      endpoints.push(endpoint);
+      const path = endpoint.getPath();
+      this.log.info(`Instantiated GetConsortiumEndpointV1 at ${path}`);
+    }
+    {
+      const options = { keyPairPem, consortiumRepo, plugin: this };
+      const endpoint = new GetNodeJwsEndpoint(options);
+      const path = endpoint.getPath();
+      endpoints.push(endpoint);
+      this.log.info(`Instantiated GetNodeJwsEndpoint at ${path}`);
+    }
+    {
+      const opts: IGetPrometheusExporterMetricsEndpointV1Options = {
+        plugin: this,
+        logLevel: this.options.logLevel,
+      };
+      const endpoint = new GetPrometheusExporterMetricsEndpointV1(opts);
+      const path = endpoint.getPath();
+      endpoints.push(endpoint);
+      this.log.info(`Instantiated GetNodeJwsEndpoint at ${path}`);
+    }
+    this.endpoints = endpoints;
+
+    log.info(`Instantiated web svcs for plugin ${pkgName} OK`, { endpoints });
+    return endpoints;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cacti-plugin-consortium-static`;
+  }
+
+  public async getNodeJws(): Promise<JWSGeneral> {
+    Checks.truthy(this.repo, `${this.className}.this.repo`);
+    const { keyPairPem } = this.options;
+
+    this.updateMetricNodeCount();
+    const keyPair = await importPKCS8(keyPairPem, "ES256K");
+    const payloadObject = { consortiumDatabase: this.repo.consortiumDatabase };
+    const payloadJson = safeStableStringify(payloadObject);
+    const _protected = {
+      iat: Date.now(),
+      jti: uuidv4(),
+      iss: "Hyperledger Cactus",
+    };
+    // TODO: double check if this casting is safe (it is supposed to be)
+    const encoder = new TextEncoder();
```

Comment:
please fix this. Installing the types for jose as a dev dependency might help. see https://www.npmjs.com/package/@types/node-jose

Reply:
This piece of code is actually from the ConsortiumManual plugin, and the comment from peter so I am not sure what is the possible issue here. 
@petermetz any notes on this one? Maybe it's a comment left behind ...

Reply:
@eduv09 @RafaelAPB Sorry for the slow processing! 
Short answer: I think we are OK to leave this as-is for now and I'll help with some work on the longer term solution that guarantees correct validation and eliminates the need for casting altogether. 

Longer answer:
I looked into it in detail and the way I'd recommend handling it is writing some type-narrowing user-defined type guards (a Typescript language feature) within the core-api package itself and then use those to verify both at build and runtime that the object's are valid and are matching the schema of the openapi spec.

I'm pasting a rough example here which is not yet operational but is the draft idea I'll send a quick PR with soon.

While I do that PR this should get merged because it doesn't have a huge impact and I definitely don't want to hold up the PR with putting a dependency on it.

The draft code I'll use to get started with my improvements (no need to do anything with it I'm just taking it as notes for self mostly)

```typescript
import Ajv from 'ajv';

const ajv = new Ajv();

// JWSCompact type guard
function isJWSCompact(data: unknown): data is string {
  const schema = {
    type: 'string',
    minLength: 5,
    maxLength: 65535,
    pattern: /^[a-zA-Z0-9-_]+?.[a-zA-Z0-9-_]+?.([a-zA-Z0-9-_]+)?$/,
  };

  const validate = ajv.compile(schema);
  return validate(data);
}

// JWSRecipient type guard
function isJWSRecipient(data: unknown): data is JWSRecipient {
  const schema = {
    type: 'object',
    required: ['signature'],
    properties: {
      signature: { type: 'string' },
      protected: { type: 'string' },
      header: { type: 'object', additionalProperties: true },
    },
  };

  const validate = ajv.compile(schema);
  return validate(data);
}

// JWSGeneral type guard (interface for JWSRecipient)
interface JWSRecipient {
  signature: string;
  protected?: string;
  header?: Record<string, unknown>;
}

function isJWSGeneral(data: unknown): data is JWSGeneral {
  const schema = {
    type: 'object',
    required: ['payload', 'signatures'],
    properties: {
      payload: { type: 'string', minLength: 1, maxLength: 65535 },
      signatures: {
        type: 'array',
        items: { $ref: '#/components/schemas/JWSRecipient' },
      },
    },
  };

  const validate = ajv.compile(schema);
  return validate(data);
}

export { isJWSCompact, isJWSRecipient, isJWSGeneral };
```